src/mock_miner.cpp

```diff
@@ -20,12 +20,13 @@ bool mineBlocks(const uint32_t& n,
                 std::shared_ptr<BlockTree<Block, BlockParams>> blockChain,
                 ValidationState& state) {
   bool res = true;
+  BlockIndex<Block> temp_index;
   for (uint32_t i = 0; i < n && res; ++i) {
     BlockIndex<Block>* tip = blockChain->getBestChain().tip();
     assert(tip != nullptr);
 
     Block block = miner->createNextBlock(*tip, {});
-    res = blockChain->acceptBlock(block, state);
+    res = blockChain->acceptBlock(block, temp_index, state);
```

Comment:
Move BlockIndex<Block> temp_index inside the loop - decrease it's scope.
Explicitly ignore result, eg, (void)temp_index;